File: frontend/src/types/user_session.ts
/**
 * User Session Type Definitions
 * 
 * This file contains all TypeScript types for managing user sessions
 * for persistence and API operations.
 */

import { Mission } from '@/types/workflow';
import { Chat } from '@/types/chat';

export enum UserSessionStatus {
    ACTIVE = 'active',
    COMPLETED = 'completed',
    ABANDONED = 'abandoned',
    ARCHIVED = 'archived'
}

// Core session entity
export interface UserSession {
    id: string;
    user_id: number;
    name?: string;
    status: UserSessionStatus;
    session_metadata: Record<string, any>;
    created_at: string;
    updated_at: string;
    last_activity_at: string;

    // Relationships (populated by services)
    chat?: Chat;
    mission?: Mission;
}

// API Request/Response types
export interface CreateUserSessionRequest {
    name?: string;
    session_metadata?: Record<string, any>;
}

export interface CreateUserSessionResponse {
    user_session: UserSession;
    chat: Chat;
}

// Backend response format - lightweight response with just IDs
export interface CreateUserSessionBackendResponse {
    id: string;
    user_id: number;
    name?: string;
    chat_id: string;
    mission_id?: string;
    session_metadata: Record<string, any>;
}

export interface UpdateUserSessionRequest {
    name?: string;
    status?: UserSessionStatus;
    session_metadata?: Record<string, any>;
}

export interface ListUserSessionsResponse {
    sessions: UserSession[];
    total: number;
    page: number;
    per_page: number;
}

// Summary model for efficient listing
export interface UserSessionSummary {
    id: string;
    user_id: number;
    name?: string;
    status: UserSessionStatus;
    created_at: string;
    updated_at: string;
    last_activity_at: string;

    // Summary info
    message_count: number;
    has_mission: boolean;
} 

File: backend/schemas/user_session.py
"""
User Session Schema Definitions

This module contains all Pydantic models for managing user sessions
for persistence and API operations.
"""

from typing import Optional, Any, Dict
from pydantic import BaseModel, Field
from datetime import datetime
from enum import Enum
from .workflow import Mission


class UserSessionStatus(str, Enum):
    """Status of a user session"""
    ACTIVE = "active"
    COMPLETED = "completed"
    ABANDONED = "abandoned"
    ARCHIVED = "archived"


# Core session persistence model
class UserSession(BaseModel):
    """User session representing a workspace/conversation container"""
    id: str = Field(description="Unique identifier for the session")
    user_id: int = Field(description="ID of the user who owns this session")
    name: str = Field(description="Name/title of the session")
    status: UserSessionStatus = Field(default=UserSessionStatus.ACTIVE, description="Current status of the session")
    
    # Relationships
    chat_id: str = Field(description="ID of the associated chat")
    mission_id: Optional[str] = Field(default=None, description="ID of the associated mission if created")
    
    # Metadata
    session_metadata: Dict[str, Any] = Field(default_factory=dict, description="Additional session metadata")
    created_at: datetime = Field(description="When the session was created")
    updated_at: datetime = Field(description="When the session was last updated")
    last_activity_at: datetime = Field(description="When the session had its last activity")
    
    # Relationships (populated by services)
    chat: Optional['Chat'] = Field(default=None, description="Associated chat conversation")
    mission: Optional[Mission] = Field(default=None, description="Associated mission if created")


# Service Response models (used by service layer)
class CreateUserSessionResponse(BaseModel):
    """Response when creating a new user session - used by service layer"""
    user_session: UserSession = Field(description="Created user session")
    chat: 'Chat' = Field(description="Associated chat created with the session")

# Import Chat for forward references
from .chat import Chat 